feat(common) : 공통 기능 작성

Ko-udon opened the pull request:
## 변경사항

<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
📌 AS-IS
프로젝트 생성

📌 TO-BE
1. h2-console
    - application.properties
    - url, name, password, 쿼리 출력 설정



2. BaseEntity
    - Envers gradle 추가
    - erd 공통 데이터로 생성일(createdAt), 수정일(modifiedAt) 추가



3. Swagger
    - SwaggerConfig 추가



4. logback
    - logback-spring.xml 추가
        - log level : INFO
        - 로그 파일 저장 방식 (일반로그/에러로그)



5. GlobalExceptionHandler
    - 글로벌 단위에서의 예외처리 핸들러 추가
    - 글로벌한 예외처리 내용으로 GlobalErrorCode 작성



## 체크 리스트 (Checklist)

pull request 전에 아래 체크 리스트들을 만족하는 지 확인한 후 체크('x') 표시를 해주시기 바랍니다.

- [x]  master 브랜치가 아니라 **develop** 브랜치에 Merge하도록 pull request를 작성 중이신가요?
- [x]  모든 **테스트**가 성공했나요?

ehddn committed:
feat(common) : 공통 기능 작성

1. h2-console
- application.properties
  - url, name, password, 쿼리 출력 설정

2. BaseEntity
- Envers gradle 추가
- erd 공통 데이터로 생성일(createdAt), 수정일(modifiedAt) 추가

3. Swagger
- SwaggerConfig 추가

4. logback
- logback-spring.xml 추가
  - log level : INFO
  - 로그 파일 저장 방식 (일반로그/에러로그)

5.
GlobalExceptionHandler
- 글로벌 단위에서의 예외처리 핸들러 추가
- 글로벌한 예외처리 내용으로
GlobalErrorCode 작성

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/entity/BaseEntity.java:
```diff
@@ -0,0 +1,20 @@
+
+@Getter
+@MappedSuperclass
+@EntityListeners(value = {AuditingEntityListener.class})
```
@EntityListeners 이 동작하려면 @EnableJpaAuditing 설정이 필요한 걸로 알고 있어요~
JpaAuditingConfig.java를 따로 만들던가 WithMarketApplication.java 위에 Enable 관련은 모아두는 방식이 고려되네요

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Getter
+    @AllArgsConstructor
+    public static class ExceptionResponse {
```
ExceptionResponse는 구별되기 쉽게 밖으로 빼는 게 좋을 거 같아요

Kim-SeonBeom replied:
수정완료했습니다

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/exception/GlobalErrorCode.java:
```diff
@@ -0,0 +1,25 @@
+
+@RequiredArgsConstructor
+@Getter
+public enum GlobalErrorCode {
```
GlobalErrorCode는 이제 domain별로 ErrorCode를 가져갈 거기 때문에 interface로 만드는 게 좋을거 같네요
interface로 만들어두고 각각 domain에서 implements 받아서 사용하면 될거 같네요

Ko-udon commented on src/main/java/com/zerobase/wishmarket/entity/BaseEntity.java:
```diff
@@ -0,0 +1,20 @@
+
+@Getter
+@MappedSuperclass
+@EntityListeners(value = {AuditingEntityListener.class})
```
수정 예정입니다~

Kim-SeonBeom replied:
확인했습니다~

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/config/SwaggerConfig.java:
```diff
@@ -0,0 +1,35 @@
+
+@Slf4j
+@Configuration
+public class SwaggerConfig {
```
@EnableSwagger2 관련 설정이 빠진 거 같아요

paik-dongjoo commented on src/main/java/com/zerobase/wishmarket/exception/GlobalErrorCode.java:
```diff
@@ -0,0 +1,25 @@
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
```
review 작동 테스트

ehddn committed:
refactor(공통 부분 설정 수정):

1. SwaggerConfig
 - @EnableSwagger2 추가

2. @EnableJpaAuditing 메인 Application에 추가

3. Exception 관련 코드 수정

seyoung5744 commented:
commit을 이런 식으로 작성하는 게 맞는거 같네요
refactor(common):엉뚱한 부분 설정 수정

seyoung5744 reviewed (approved):
확인 했습니다. 고생하셨습니다~

Ko-udon commented:
모두 수정완료하였습니다~

Kim-SeonBeom commented on src/main/java/com/zerobase/wishmarket/WishMarketApplication.java:
```diff
@@ -0,0 +1,14 @@
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
+@EnableJpaAuditing
```
JpaAuditing annotation 추가 완료

Kim-SeonBeom commented on build.gradle:
```diff
@@ -21,9 +21,17 @@ repositories {
+
+    implementation 'org.springframework.data:spring-data-envers'
+    implementation 'io.springfox:springfox-boot-starter:3.0.0'
+    implementation 'io.springfox:springfox-swagger-ui:3.0.0'
```
수정완료했습니다
